source/opt/eliminate_dead_constant_pass.cpp

```diff
@@ -91,27 +91,10 @@ Pass::Status EliminateDeadConstantPass::Process(ir::IRContext* irContext) {
     working_list.erase(inst);
   }
 
-  // Find all annotation and debug instructions that are referencing dead
-  // constants.
-  std::unordered_set<ir::Instruction*> dead_others;
-  for (auto* dc : dead_consts) {
-    irContext->get_def_use_mgr()->ForEachUser(
-        dc, [&dead_others](ir::Instruction* user) {
-          SpvOp op = user->opcode();
-          if (ir::IsAnnotationInst(op) || ir::IsDebug1Inst(op) ||
-              ir::IsDebug2Inst(op) || ir::IsDebug3Inst(op)) {
-            dead_others.insert(user);
-          }
-        });
-  }
-
```

Comment:
OK, so I probably missed something earlier.  Why is it ok to remove this code now?

Reply:
This is essentially doing the same thing as "KillNameAndDecorations", which is done as part of Killinst.  So the loop below that was removed was simply trying to remove instructions that have already been removed.